dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractAnalyticsService.java

```diff
@@ -383,20 +387,28 @@ protected void addMetadata(
 
       if (hasResults) {
         optionItems.addAll(
-            optionsPresentInGrid.values().stream()
-                .flatMap(Collection::stream)
-                .distinct()
-                .collect(toList()));
+            optionsPresentInGrid.values().stream().flatMap(Collection::stream).distinct().toList());
       } else {
         optionItems.addAll(getItemOptionsAsFilter(params.getItemOptions(), params.getItems()));
       }
 
+      Map<String, Object> items = new HashMap<>();
+      AnalyticsOrganisationUnitUtils.getUserOrganisationUnitsUidList(
+              currentUserService.getCurrentUser(), params.getUserOrganisationUnitsCriteria())
+          .forEach(items::putAll);
+
+      if (params.isComingFromQuery()) {
```

Comment:
Could we concentrate all the logic of the if/else related to
```
params.isComingFromQuery()
```
on the same block? ie:

```
      if (params.isComingFromQuery()) {
        items.putAll(getMetadataItems(params, periodKeywords, optionItems, grid));
        metadata.put(
            DIMENSIONS.getKey(), getDimensionItems(params, Optional.of(optionsPresentInGrid)));
      } else {
        items.putAll(getMetadataItems(params));
        metadata.put(DIMENSIONS.getKey(), getDimensionItems(params, empty()));
      }

      metadata.put(ITEMS.getKey(), items);
```
Or is it going to have a side affect?

Reply:
the items will change the place with dimensions, I thing this should not be a problem, should I change it?